### DISCUSSION TITLE
Cannot find module '@repo/xxxx' or its corresponding type declarations.

### DISCUSSION THREAD

**Discussion by lstoryc** [vercel/turborepo#8665] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

https://github.com/vercel/turbo/issues/new?assignees=&labels=kind%3A+bug%2Cowned-by%3A+turborepo%2Cneeds%3A+triage&projects=

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Linux

### Which canary version will you have in your reproduction?

turbo@2.0.1

### Describe the Bug

when I use in local. everthing is ok。
But it's not working when I build on linux server.

Below is the build log, 

----
pnpm run build-develop v1
commitId de96b4e
/workspace/source

added 1 package in 1s

1 package is looking for funding
  run `npm fund` for details
v18.16.1
8.15.8
Scope: all 8 workspace projects
Lockfile is up to date, resolution step is skipped
Progress: resolved 1, reused 0, downloaded 0, added 0
Packages: +1704
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Progress: resolved 1704, reused 0, downloaded 0, added 0
Progress: resolved 1704, reused 0, downloaded 211, added 0
Progress: resolved 1704, reused 0, downloaded 303, added 84
Progress: resolved 1704, reused 0, downloaded 324, added 297
Progress: resolved 1704, reused 0, downloaded 389, added 360
Progress: resolved 1704, reused 0, downloaded 437, added 408
Progress: resolved 1704, reused 0, downloaded 495, added 477
Progress: resolved 1704, reused 0, downloaded 525, added 505
Progress: resolved 1704, reused 0, downloaded 574, added 547
Progress: resolved 1704, reused 0, downloaded 600, added 581
Progress: resolved 1704, reused 0, downloaded 627, added 604
Progress: resolved 1704, reused 0, downloaded 666, added 635
Progress: resolved 1704, reused 0, downloaded 696, added 666
Progress: resolved 1704, reused 0, downloaded 714, added 688
Progress: resolved 1704, reused 0, downloaded 724, added 703
Progress: resolved 1704, reused 0, downloaded 736, added 715
Progress: resolved 1704, reused 0, downloaded 753, added 728
Progress: resolved 1704, reused 0, downloaded 759, added 734
Progress: resolved 1704, reused 0, downloaded 773, added 752
Progress: resolved 1704, reused 0, downloaded 790, added 770
Progress: resolved 1704, reused 0, downloaded 806, added 784
Progress: resolved 1704, reused 0, downloaded 836, added 814
Progress: resolved 1704, reused 0, downloaded 860, added 836
Progress: resolved 1704, reused 0, downloaded 885, added 860
Progress: resolved 1704, reused 0, downloaded 914, added 885
Progress: resolved 1704, reused 0, downloaded 965, added 927
Progress: resolved 1704, reused 0, downloaded 1011, added 976
Progress: resolved 1704, reused 0, downloaded 1053, added 1016
Progress: resolved 1704, reused 0, downloaded 1087, added 1057
Progress: resolved 1704, reused 0, downloaded 1126, added 1088
Progress: resolved 1704, reused 0, downloaded 1183, added 1143
Progress: resolved 1704, reused 0, downloaded 1263, added 1234
Progress: resolved 1704, reused 0, downloaded 1337, added 1298
Progress: resolved 1704, reused 0, downloaded 1420, added 1382
Progress: resolved 1704, reused 0, downloaded 1514, added 1474
Progress: resolved 1704, reused 0, downloaded 1642, added 1609
Progress: resolved 1704, reused 0, downloaded 1692, added 1686
Progress: resolved 1704, reused 0, downloaded 1693, added 1695
Progress: resolved 1704, reused 0, downloaded 1694, added 1695
Progress: resolved 1704, reused 0, downloaded 1694, added 1698
Progress: resolved 1704, reused 0, downloaded 1694, added 1699
Progress: resolved 1704, reused 0, downloaded 1694, added 1700
Progress: resolved 1704, reused 0, downloaded 1694, added 1701
Progress: resolved 1704, reused 0, downloaded 1695, added 1702
Progress: resolved 1704, reused 0, downloaded 1695, added 1703
Progress: resolved 1704, reused 0, downloaded 1695, added 1704
Progress: resolved 1704, reused 0, downloaded 1695, added 1704, done
.../node_modules/core-js-pure postinstall$ node -e "try{require('./postinstall')}catch(e){}"
.../core-js@3.33.2/node_modules/core-js postinstall$ node -e "try{require('./postinstall')}catch(e){}"
.../node_modules/core-js-pure postinstall: Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!
.../node_modules/core-js-pure postinstall: The project needs your help! Please consider supporting core-js:
.../node_modules/core-js-pure postinstall: > https://opencollective.com/core-js 
.../node_modules/core-js-pure postinstall: > https://patreon.com/zloirock 
.../node_modules/core-js-pure postinstall: > https://boosty.to/zloirock 
.../node_modules/core-js-pure postinstall: > bitcoin: bc1qlea7544qtsmj2rayg0lthvza9fau63ux0fstcz 
.../node_modules/core-js-pure postinstall: I highly recommend reading this: https://github.com/zloirock/core-js/blob/master/docs/2023-02-14-so-whats-next.md 
.../node_modules/core-js-pure postinstall: Done
.../core-js@2.6.12/node_modules/core-js postinstall$ node -e "try{require('./postinstall')}catch(e){}"
.../core-js@3.33.2/node_modules/core-js postinstall: Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!
.../core-js@3.33.2/node_modules/core-js postinstall: The project needs your help! Please consider supporting core-js:
.../core-js@3.33.2/node_modules/core-js postinstall: > https://opencollective.com/core-js 
.../core-js@3.33.2/node_modules/core-js postinstall: > https://patreon.com/zloirock 
.../core-js@3.33.2/node_modules/core-js postinstall: > https://boosty.to/zloirock 
.../core-js@3.33.2/node_modules/core-js postinstall: > bitcoin: bc1qlea7544qtsmj2rayg0lthvza9fau63ux0fstcz 
.../core-js@3.33.2/node_modules/core-js postinstall: I highly recommend reading this: https://github.com/zloirock/core-js/blob/master/docs/2023-02-14-so-whats-next.md 
.../core-js@3.33.2/node_modules/core-js postinstall: Done
.../yorkie@2.0.0/node_modules/yorkie install$ node bin/install.js
.../core-js@2.6.12/node_modules/core-js postinstall: Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!
.../core-js@2.6.12/node_modules/core-js postinstall: The project needs your help! Please consider supporting of core-js on Open Collective or Patreon: 
.../core-js@2.6.12/node_modules/core-js postinstall: > https://opencollective.com/core-js 
.../core-js@2.6.12/node_modules/core-js postinstall: > https://www.patreon.com/zloirock 
.../core-js@2.6.12/node_modules/core-js postinstall: Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)
.../core-js@2.6.12/node_modules/core-js postinstall: Done
.../node_modules/vue-popperjs postinstall$ opencollective-postinstall || true
.../yorkie@2.0.0/node_modules/yorkie install: setting up Git hooks
.../yorkie@2.0.0/node_modules/yorkie install: trying to install from sub 'node_module' directory, skipping Git hooks installation
.../yorkie@2.0.0/node_modules/yorkie install: Done
.../node_modules/vue-popperjs postinstall: [96m[1mThank you for using vue-popperjs![96m[1m
.../node_modules/vue-popperjs postinstall: [0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
.../node_modules/vue-popperjs postinstall: > [94mhttps://opencollective.com/vue-popper/donate[0m
.../node_modules/vue-popperjs postinstall: Done

devDependencies:
+ @repo/eslint-config 0.0.0 <- packages/eslint-config
+ @repo/typescript-config 0.0.0 <- packages/typescript-config
+ @types/node 20.14.8
+ prettier 3.3.0
+ turbo 2.0.1

Done in 1m 7.5s

> kwebclient_webchart_monorepo@ build-develop /workspace/source
> turbo telemetry disable && turbo build:admin build:develop --  "v1"

Attention:
Turborepo now collects completely anonymous telemetry regarding usage.
This information is used to shape the Turborepo roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://turbo.build/repo/docs/telemetry

Success!

Status: Disabled

You have opted-out of Turborepo anonymous telemetry. No data will be collected from your machine.
Learn more: https://turbo.build/repo/docs/telemetry
• Packages in scope: @repo/common, @repo/eslint-config, @repo/sdk, @repo/share, @repo/typescript-config, chartproduct, client-pages
• Running build:admin, build:develop in 7 packages
• Remote caching disabled
@repo/common:build:common: cache miss, executing 5dacef052a0cf26d
@repo/sdk:build:sdk: cache miss, executing 5422d9ddff08174f
client-pages:build:develop: cache miss, executing d279e5bfcf864e38
chartproduct:build:admin: cache miss, executing 0e481077264b12c7
@repo/common:build:common: 
@repo/common:build:common: > @repo/common@0.0.1 build:common /workspace/source/packages/common
@repo/common:build:common: > pnpm run build:types && pnpm run build:common-client && pnpm run build:common-web
@repo/common:build:common: 
client-pages:build:develop: 
client-pages:build:develop: > client-pages@1.0.0 build:develop /workspace/source/apps/client-pages
client-pages:build:develop: > vue-cli-service build --mode development "v1"
client-pages:build:develop: 
@repo/sdk:build:sdk: 
@repo/sdk:build:sdk: > @repo/sdk@0.0.1 build:sdk /workspace/source/packages/sdk
@repo/sdk:build:sdk: > pnpm run build:types && pnpm run build:client && pnpm run build:web
@repo/sdk:build:sdk: 
chartproduct:build:admin: 
chartproduct:build:admin: > chartproduct@1.0.0 build:admin /workspace/source/apps/chart-product
chartproduct:build:admin: > vue-cli-service build --mode development_chartproduct "v1"
chartproduct:build:admin: 
@repo/common:build:common: 
@repo/common:build:common: > @repo/common@0.0.1 build:types /workspace/source/packages/common
@repo/common:build:common: > tsc
@repo/common:build:common: 
@repo/sdk:build:sdk: 
@repo/sdk:build:sdk: > @repo/sdk@0.0.1 build:types /workspace/source/packages/sdk
@repo/sdk:build:sdk: > tsc
@repo/sdk:build:sdk: 
client-pages:build:develop: getPublicPath dev //web.docer.wpscdn.cn/docer/fe/wps_pc_client_hybrid_fe-kwebclient_webchart-dist/v1/
chartproduct:build:admin: getPublicPath chartproduct //web.docer.wpscdn.cn/docer/fe/wps_pc_client_hybrid_fe-kwebclient_webchart-dist/dashborad/v1/
client-pages:build:develop: getOutputDir dev ../../dist/v1
chartproduct:build:admin: getOutputDir chartproduct ../../dist/dashborad/v1
client-pages:build:develop: All browser targets in the browserslist configuration have supported ES module.
client-pages:build:develop: Therefore we don't build two separate bundles for differential loading.
client-pages:build:develop: 
chartproduct:build:admin: All browser targets in the browserslist configuration have supported ES module.
chartproduct:build:admin: Therefore we don't build two separate bundles for differential loading.
chartproduct:build:admin: 
chartproduct:build:admin: 
client-pages:build:develop: 
client-pages:build:develop: -  Building for development...
chartproduct:build:admin: -  Building for development_chartproduct...
@repo/common:build:common: 
@repo/common:build:common: > @repo/common@0.0.1 build:common-client /workspace/source/packages/common
@repo/common:build:common: > rollup -c rollup.config.ts --configPlugin typescript --environment BUILD:client
@repo/common:build:common: 
@repo/sdk:build:sdk: src/data-source/highlight.ts(3,47): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/date-axis.ts(1,44): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/date-axis.ts(5,36): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-2d-area.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-2d-base.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-2d-doughnut.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-2d-line.ts(5,26): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-2d-pictorial.ts(4,34): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-2d-pie.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-2d-radar.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-2d-rose.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-2d-scatter.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-2d-waterfall.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-3d-column.ts(1,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-base.ts(4,8): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-base.ts(11,8): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-base.ts(12,50): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-base.ts(37,33): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-base.ts(43,23): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-base.ts(45,52): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-funnel.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-gauge.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-gauge.ts(6,33): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-liquidfill.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-liquidfill.ts(4,33): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-sankey.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-sankey.ts(4,33): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-treemap.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-treemap.ts(4,33): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-wordcloud.ts(3,30): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/echarts-wordcloud.ts(4,33): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/utils/chart-util.ts(15,43): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk: src/value-axis.ts(1,23): error TS2307: Cannot find module '@repo/common' or its corresponding type declarations.
@repo/sdk:build:sdk:  ELIFECYCLE  Command failed with exit code 2.
@repo/sdk:build:sdk:  ELIFECYCLE  Command failed with exit code 2.
@repo/sdk:build:sdk: ERROR: command finished with error: command (/workspace/source/packages/sdk) /usr/local/bin/pnpm run build:sdk exited (2)
@repo/sdk#build:sdk: command (/workspace/source/packages/sdk) /usr/local/bin/pnpm run build:sdk exited (2)

 Tasks:    0 successful, 4 total
Cached:    0 cached, 4 total
  Time:    8.577s 
Failed:    @repo/sdk#build:sdk

 ERROR  run failed: command  exited (2)
 ELIFECYCLE  Command failed with exit code 2.
构建失败
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   237  100    15  100   222     52    781 --:--:-- --:--:-- --:--:--   837
{"result":"ok"}
----

the Trubo.json is 
----
{
  "$schema": "https://turbo.build/schema.json",
  "globalDependencies": [
    "**/.env.*"
  ],
  "tasks": {
    "build:client": {
      "dependsOn": [
        "build:sdk"
      ],
      "outputs": [
        "dist/**"
      ]
    },
    "build:admin": {
      "dependsOn": [
        "build:sdk"
      ],
      "outputs": [
        "dist/**"
      ]
    },
    "build:develop": {
      "dependsOn": [
        "build:sdk"
      ],
      "outputs": [
        "dist/**"
      ]
    },
    "build:release": {
      "dependsOn": [
        "build:sdk"
      ],
      "outputs": [
        "dist/**"
      ]
    },
    "build:rc": {
      "dependsOn": [
        "build:sdk"
      ],
      "outputs": [
        "dist/**"
      ]
    },
    "build:common": {
      "outputs": [
        "dist/**"
      ]
    },
    "build:sdk": {
      "dependsOn": [
        "build:common"
      ],
      "outputs": [
        "dist/**"
      ]
    },
    "lint": {
      "dependsOn": [
        "^lint"
      ]
    },
    "lint:admin": {
      "dependsOn": [
        "^lint"
      ]
    },
    "dev:client": {
      "cache": false,
      "persistent": true,
      "dependsOn": [
        "build:sdk"
      ]
    },
    "dev:admin": {
      "cache": false,
      "persistent": true,
      "dependsOn": [
        "build:sdk"
      ]
    }
  }
}
----

### Expected Behavior

build success

### To Reproduce

nothing

**Comment by anthonyshew** [comment]
I'll move this to a Discussion since it's highly likely to be a user misconfiguration with TypeScript rather than a Turborepo bug.

**Comment by anthonyshew** [comment]
Could you reformat your post so that it's readable? You're much more likely to get help if you make it easer for the community to help you. Additionally, do you have a reproduction folks can look at? TypeScript issues can come from multiple different places so it can be easier for others to debug with the code available.

It's likely that it works locally because the needed files are already on disk before you run the `turbo build` command. My guess is that, if you were to delete all your `dist` folders and then run `turbo build` from clean, you'd see the same error.

However, I'm only guessing that that might be the case since it's hard to tell. More information would be helpful.

**Reply by lstoryc** [reply]
When I use below command to run , it works fine . so I belive that the problem is cause by **concurrency**. when I set concurrency=2, it happen again

`turbo x1 x2 --concurrency=1`
